File: Python/Other algo/sliding_window.py
def maxSUmSubarray(arr, num):
    maxSum = 0
    tempSum = 0
    if len(arr) < num:
        return None
    for i in range(0,num):
        tempSum += arr[i]
    maxSum = tempSum
    for i in range(num,len(arr)):
        tempSum = tempSum - arr[i-num] + arr[i]
        maxSum = max(tempSum,maxSum)
    return maxSum

print(maxSUmSubarray([1,4,6,2,7,1,6],3))

File: Js/Other Algo/Sliding_window.js
function maxSumSubarray(arr,num){
  var maxSum = 0;
  var tempSum = 0;
  if(arr.length < num) return null;
  for(let i=0;i<num;i++){
    maxSum += arr[i];
  }
  tempSum = maxSum;
  for(let i=num;i<arr.length;i++){
    tempSum = tempSum - arr[i-num] + arr[i];
    maxSum = Math.max(tempSum,maxSum);
  }
  return maxSum;
}

console.log(maxSumSubarray([2,3,4,6,5,1,6,2],3))